### DISCUSSION TITLE
Use of the software. Database merge.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#123] [Q&A] [answered]
the more I use it the more I appreciate it : really nice work

I remind you that we are volunteers so we manage the library as we can
I would like to understand the pagination because it can bring us a more efficient way to manage

practical case, I join you my work base

I look for possible errors in the spelling of the editors

I put in page, I sort on the column editor and I look at the possible error (typo, bad characters ... our base is really "dirty").

I give you an example with glena and someone has entered gelna in the database

how do you proceed with the pagination ?

[test-modif-perso-not-null.sqlite.tar.gz](https://github.com/textbrowser/biblioteq/files/7514393/test-modif-perso-not-null.sqlite.tar.gz)

**Comment by textbrowser** [comment]
I see. That's a visual inspection unless you export the contents of the current view. I would categorize this under a tool. BQ should not be a spell checker. Also remember that you have access to the SQLite database.

Here is an example of using SQLite's dump feature.

https://www.sqlitetutorial.net/sqlite-dump/

So, you dump the contents of the database and insert them into your favorite editor. For example, Libre Office. That tool has a spell checker and you can then search the entire text for spelling errors. You can carefully correct them and then either import the results into another SQLite database and/or apply the corrections through BQ.

**Comment by textbrowser** [comment]
I don't know the preferred approach. A spell checker is interesting but a lot of work. It then burdens maintenance and duplication. Like, I don't think of BiblioteQ as the proverbial Swiss Army knife.

**Comment by meteos77** [comment]
it's nice to share your way of doing things
I will test

1 - no spelling approach required
1bis - no spelling approach will be asked

2 - some people in the library don't even know spreadsheets so the term SQL is not the level.

3 - at least some of the Swiss speak French :-)

**Reply by textbrowser** [reply]
Software is like... learning new. Some software requires that you learn much more than just about it. It's not a closed system.

**Reply by meteos77** [reply]
especially with free software that allows you to put your nose under the hood :-)
how will you translate this one : I prefer to play with the toys in the house rather than having the windows closed :-)

A very big THANK YOU again for this, it's very instructive and if I would never develop software, I think I will do my part by trying to make feedback, translations and proposals for evolutions.

**Comment by meteos77** [comment]
after test on spell checker, it seems to me unfortunately not adapted that would have been too beautiful :)

the accents of the French language, the proper names of the titles of the books, the puns, finally XXX exceptions make that one has more quickly made to open LibreOffice is to seize the infos.

I have in mind to rather ask the volunteer to go and enter the ISBN code with the shower head and then once we have the database with the books completed on the field N° ISBN go to a site that accepts to enter xxx isbn request and get out the complete records for the books.

after a small script to take the internal barcode number of the old file to the new data of the book should I think allow us to have a fairly square base. without too much input.

**Comment by textbrowser** [comment]
Whenever I compose a document where a spell checker is not available within the thing I am composing, I refer to a separate tool. There is a correlation between software and reality. Of course software has the ability to be adaptable more so than reality. This does not necessarily imply that this adaptability is free of work. BiblioteQ is not an editor.

Order of importance?

1. Scanner.
2. Importer.
3. Spell checker.

**Comment by meteos77** [comment]
1 - the unimarc import and the 995 fields is an obligation without it no library for us so no software to manage the library.

2 - output of books with showerhead and return of books with showerhead


3 - .... many other improvements

version 1000000 - Spell checker: I will never consider BiblioteQ as an editor, there is only the fact that during the volunteers' permanence they can find a misspelled information, it opens the document file and completes it with the new information. Today this seems to me to be handled perfectly.

**Comment by textbrowser** [comment]
Agreed.

I think of your spell-checking proposition as this. Yes, I would enjoy such a feature in BQ. Then I think of portability. For example, I integrated the Poppler library some years ago. On Unix-like systems, this was easy to maintain. On Windows, not so. So, it became lost on Windows. It feels incomplete now. Similar to YAZ on Android. YAZ, however, has two other replacements. With spelling, dictionaries are necessary.

We also have the issue of simplifying the loan process.

**Comment by meteos77** [comment]
Hello,
I think there is a misunderstanding because I am not talking about a spell check at all?

I was explaining how you could use the main view with all articles to correct spelling but all manually. No automation or development at all on your part. (at least for our use).

**Comment by textbrowser** [comment]
No, I understand. Simple features require complex implementations. Like, spell-checking is simple.

**Comment by meteos77** [comment]
simple with the French language ahah :-) it's antinomic
take the csv file https://github.com/textbrowser/biblioteq/discussions/119#discussioncomment-1614167
 I sent you from the current database and do the spell check in Libreoffice with French spelling. 

well if you want to implement it, why not but it's not a priority for me and you told me to concentrate on the main requests, I try to follow your advice :-)
well if in 2 hours you manage to make a clean csv file with your proofreader it will be great but I have a big doubt :-)

**Comment by meteos77** [comment]
exemple 
capital letter : UN INTERNE TUE. 
Considering that the letters could be accentuated, this sentence can give rise to **four different interpretations**
Un interne tue – Un interne tué – Un interné tue – Un interné tué. 

You miss one accent and everything is turned upside down!
Un interne tue = An intern kills
Un interne tué = An intern killed
Un interné tue = An inmate kills
Un interné tué = An inmate killed

in French the accent can change the meaning
then as for the titles are in capital letters, to put them back with the good interpretation

**Comment by textbrowser** [comment]
Software is not a replacement of humans. :) The responsibility lies with the writer.

**Comment by meteos77** [comment]
the entries in the base were made maybe fifteen years ago and the capital letters were not accentuated at the time (habit of the people using typewriters without the accentuated letters)


![machine](https://user-images.githubusercontent.com/56431039/141340606-75b865c8-bd5a-4ece-8158-a965f732f077.jpeg)

we will take that back afterwards it is not a big problem we manage the loans of books rather by the internal Codebare, the isbn 



anyway there is a human to manage the loan so it's ok even with the wrong spelling
in the database there are even some articles without title :-)

**Reply by textbrowser** [reply]
People collect typewriters. The typewriter featured in your image utilizes the AZERTY keyboard layout. Notice that the keys are staggered. There is an alternate arrangement, columnar.  Modern keyboards were born from typewriters because of mechanical typewriters.

https://en.wikipedia.org/wiki/Path_dependence

**Comment by meteos77** [comment]
the azerty keyboard is used in France
I learned the keyboard with a machine of this style :-)

**Comment by textbrowser** [comment]
Release notes will be available in the executable for the next release.

**Reply by meteos77** [reply]
should we translate the realease notes ?

I like the look of your page [](https://textbrowser.github.io/biblioteq/)   it reminds me the sobriety of beos :-)

**Reply by textbrowser** [reply]
Sure, if you'd like to.

**Comment by meteos77** [comment]
Today I started to look at how to have a cleaner csv, I used a site "moccam" for french book which allows to have the records by batch.
I am attaching 2 unimarc BDP files, the first one is OK, the second one has the 995 fields in the wrong place.

[exemples-unimarc-bdp-995.tar.gz](https://github.com/textbrowser/biblioteq/files/7529366/exemples-unimarc-bdp-995.tar.gz)

**Comment by textbrowser** [comment]
I will improve the reservation (checkout) process and then focus on the new import. The checkout process will benefit others. It's also easier to implement and has a clearer process. Is that fine with you?

**Comment by meteos77** [comment]
Would you like to create a new topic on the Loans/Returns interface?

**Comment by textbrowser** [comment]
Sure. There's a reminder too.

**Comment by meteos77** [comment]
hello, new suggested functionality

I have created two sqlite databases with cover images and I would like to merge them now.

this would also allow to check beforehand and not to import in the main base bad documents.

**Comment by textbrowser** [comment]
Merge two databases? Interesting request. Excellent. Let me think.

**Comment by meteos77** [comment]
herewith my 2 small bases
a purchase of new books
a donation of new books

2 people each work at home and go to the library to integrate them into the main database

[51-BiblioteQ_Bases.zip](https://github.com/textbrowser/biblioteq/files/7612008/51-BiblioteQ_Bases.zip)

**Comment by textbrowser** [comment]
SQLite itself may have means to merge two databases. Have you researched?

**Comment by meteos77** [comment]
I tried SQLiteStudio but I couldn't paste it

and I would prefer to have it in biblioteQ.

for example at the moment we have to send 500 books to the Departmental Library, I would like to be able to transfer the 500 books in a new database returnBD-2021-12 all the documents (without being a specialist)

this will keep track of what we send them
do like a loan but instead of putting the loan in a user it is a transfer to another base

I do not know if it is a mountain or a hill ...

**Comment by textbrowser** [comment]
:) You're full of requests. BQ should do everything.

**Reply by meteos77** [reply]
if I asked for all the ideas that come to my mind ...

**Comment by meteos77** [comment]
I remind you that I am looking for a software to manage the library for 15 years :-)

I try to be a user involved in the improvement of his possible software.

It's true that I make a lot of proposals but I try to answer to the management of the library as well as possible and as I am confronted with different needs, I make the developer that you are benefit from it.

I think that as a developer you appreciate my suggestions.
It helps to have feedback from users. It's possible or not but at least I give my ideas.

**Comment by textbrowser** [comment]
It is neither a hill nor a mountain. I know the solution already. The implementation remains.

**Comment by textbrowser** [comment]
Managing data is an art and a science. Neither of which are standard.

**Comment by meteos77** [comment]
the priorities are that in the same time 
the management of the loans/returns with showertte and the import unimarc + 995

but as I've been at the library for 3 days managing the release of 500 books, I've got some new requests that would make our life easier and maybe all BiblioteQ users

**Comment by textbrowser** [comment]
A new Tool menu item has been added.

I'm thinking of enabling it if one is connected to an SQLite database. One might wish the tool to be always available; that is, the resulting database is created if one does not already exist. This is another matter that is less important than the tool.

The tool will allow the merging of N databases. One selects each desired database and launches the merge. Some sort of error report is generated because there will be duplicates. Perhaps replacement option per instance? Another matter for later.

The end result is a merge database containing the data of N + 1 databases (N + 1 because the current database is the one containing the data of the other N databases).

**Comment by meteos77** [comment]
can we plan to move the elements from one base to another?
I would select the elements in a kind of basket and I say you move in a base not necessarily open in biblioteQ

**Comment by meteos77** [comment]
Today I made a notice so that the volunteers of the library begin to test biblioteQ, I hope to have feedback on the use (easy / not easy, blocking thing ...) finally for the moment it is download / installation and opening of the new database :-)

**Comment by textbrowser** [comment]
That's not a total merge. A select merge.

**Comment by textbrowser** [comment]
Which would require views of the databases.

**Reply by meteos77** [reply]
we see the export database but for the import database it is not necessary from the user's point of view

I want to send the documents from 1 to 10 that I have selected in the table view, right click and move to secondary base.
for the moment we have duplicate; delete modify; print; borrow

we add move or copy to base n° 2

**Comment by textbrowser** [comment]
Right; blind and constructed merges. The blind merge copies the data from N databases into the destination database. A constructed merge allows one to select items from N databases and deposit them into the destination database. We can ignore duplicates for now.

**Reply by meteos77** [reply]
if there are duplicates, place them in a trash database :-)

**Comment by meteos77** [comment]
suggestion : make a preference of which column we want to see in the table and in which order

a painful thing to use, it puts the columns in its own order even if you classify them as you wish and save the parameters.
I want to have the accession number column in position 1
the title in 2 ... 

like a table with 2 columns
left column the different columns available
we switch them to the right column we want to see
and arrows up and down allow to sort them in the order we want

**Comment by textbrowser** [comment]
This has already been discussed some years ago and I didn't find an interest in resolving it. You can momentarily change the order of the columns by dragging the column headers. The orders are discarded however. :)

**Reply by meteos77** [reply]
it's very annoying because you add a document and the column move is done directly
it should at least stay in the current working session

**Reply by textbrowser** [reply]
I'm not annoyed. :)

**Reply by meteos77** [reply]
recall your readme.md: Customized displays :-)

**Reply by textbrowser** [reply]
Sure, you can select the columns which you wish to see.

**Reply by textbrowser** [reply]
278. Preserve order of columns in the main table. (TBD)

**Reply by textbrowser** [reply]
Which makes it very difficult when one executes custom queries.

**Comment by textbrowser** [comment]
Collision repositories. Collision repositories sometimes require several layers. The point being is that one can focus on perfect solutions or on incremental solutions which lead to perfection. One step at a time.

**Reply by meteos77** [reply]
ok ok, I'll write down in my corner the possible improvements and will ask you for them. ( it goes to the translation ?)

but that's what the TODO is for, isn't it?

**Reply by textbrowser** [reply]
I already started with the blind merge.

**Reply by textbrowser** [reply]
You can submit requests as tickets because they would be requests and can be tagged.

**Reply by meteos77** [reply]
would you prefer that I post in the questions? (if that's what the tickets are)
I think it's more normal to go to the discussion area to get your opinion on the feasibility of my suggestions.

After that if it allows you to be well organized, it seems to me to be the essential.

**Comment by textbrowser** [comment, marked as answer]
OK. So I'll start on a blind merge. That will allow you to merge separate databases into a single one. That's a huge step. Then, I can focus on incremental improvements; because if you don't have the base you're certainly not going to have the more dedicated request.

**Reply by meteos77** [reply]
great; my forecast: I am in the process of re-numbering the books in the library (biblioteQ asks for isbn 13 and unique id) while the old one is based on a 4 digit id (the one I put in accession number, in case of failure of the showerhead). I have 8000 books to re-number by June.
So I make little by little a base by 50 or 100 documents then they will be integrated in the main base like that by copying base 1 to total base.

cela vous parait correcte ?

Translated with www.DeepL.com/Translator (free version)
